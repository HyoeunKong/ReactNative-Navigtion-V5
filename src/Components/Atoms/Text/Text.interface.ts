export interface ITextProps {
    size: number;
    font?: string;
    lineHeight?: number;
    color?: string;
    text?: string;
    textDecoration?: string;
}
