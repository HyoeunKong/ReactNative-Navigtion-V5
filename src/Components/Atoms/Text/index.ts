export { Text } from "./Text.impl";
