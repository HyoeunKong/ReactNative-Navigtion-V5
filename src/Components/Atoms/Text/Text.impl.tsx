import React from 'react';

import { Platform } from 'react-native';
import { ITextProps } from './Text.interface';

import { TextStyled } from './styles';

export const Text: React.FC<ITextProps> = ({
    size,
    lineHeight,
    font,
    color,
    text,
    textDecoration,
}) => {
    // const sizeNumber = parseFloat(size.substring(0, size.length - 2));
    const lineHeightNumber = lineHeight || size + 4;
    const fontSize = size + 2;
    const fontLineHiegiht = lineHeightNumber + 2;

    return (
        <TextStyled
            size={fontSize}
            lineHeight={fontLineHiegiht}
            font={font}
            color={color}
            textDecoration={textDecoration}>
            {text}
        </TextStyled>
    );
};
