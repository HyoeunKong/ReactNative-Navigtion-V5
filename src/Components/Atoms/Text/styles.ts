import styled from 'styled-components';
import { Text } from 'react-native';
import { ITextProps } from './Text.interface';

export const TextStyled = styled(Text)<ITextProps>`
    font-size: ${({ size }) => size};

    line-height: ${({ lineHeight }) => lineHeight};
    color: ${({ color }) => color};
    text-decoration: ${({ textDecoration, color }) =>
        textDecoration ? `${textDecoration} ${color}` : null};
`;
