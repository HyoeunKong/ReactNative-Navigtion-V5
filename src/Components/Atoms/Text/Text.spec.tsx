import React from "react";

import { shalloWithTheme } from "@Test/Helpers/render";
// import renderer from 'react-test-renderer';

import { Text } from "./Text.impl";

describe("Components", () => {
    describe("Text", () => {
        it("should match snapshot", () => {
            const wrapper = shalloWithTheme(
                <Text
                    size={15}
                    lineHeight={18}
                    text="블록체인기반 최초 기부 혁신을 희망하다"
                />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
