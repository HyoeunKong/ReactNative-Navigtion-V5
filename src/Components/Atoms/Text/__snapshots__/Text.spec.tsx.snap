// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Text should match snapshot 1`] = `
<Styled(Text)
  lineHeight="20px"
  size="17px"
>
  블록체인기반 최초 기부 혁신을 희망하다
</Styled(Text)>
`;
